src/tool/abi/main.cpp

```diff
@@ -0,0 +1,313 @@
+#include "pch.h"
+
+#include "abi_writer.h"
+#include "common.h"
+#include "metadata_cache.h"
+#include "strings.h"
+
+using namespace std::chrono;
+using namespace std::experimental::filesystem;
+using namespace std::literals;
+using namespace xlang;
+using namespace xlang::meta::reader;
+using namespace xlang::text;
+using namespace xlang::cmd;
+
+auto output_directory(reader const& args)
+{
+    auto out{ absolute(args.value("output", "abi")) };
+    create_directories(out);
+    out += path::preferred_separator;
+    return out.string();
+}
+
+void print_usage()
+{
+    puts("Usage...");
```

Comment:
need to implement

Reply:
This came up on my aborted PR. See: https://github.com/Microsoft/xlang/pull/51